students/stuartkershaw/lesson06/mailroom.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python3
+
+import pathlib
+pth = pathlib.Path('./')
+
+donors = {
+    'Anne Hathaway': {
+        'donations': [
+            4000,
+            6250
+        ],
+    },
+    'Geronimo Jackson': {
+        'donations': [
+            150,
+            400,
+            1250
+        ]
+    },
+    'Dingo Dogson': {
+        'donations': [
+            150,
+            300,
+            100
+        ]
+    },
+    'Leo Kottke': {
+        'donations': [
+            75,
+            125
+        ]
+    },
+    'Flask McCreole': {
+        'donations': [
+            1000
+        ]
+    },
+    'Piglet Norquist': {
+        'donations': [
+            20,
+            15,
+            30
+        ]
+    }
+}
+
+
+def compose_email(donor):
+    message_obj = {
+        'donor_name': donor,
+        'donation': donors[donor]['donations'][-1]
+    }
+    message = 'Dear {donor_name}, thanks so much '\
+              'for your generous donation in the amount of: '\
+              '${donation}.'.format(**message_obj)
+    return message
+
+
+def apply_selection(selection):
+    arg_dict = {
+        '1': send_thank_you,
+        '2': show_donor_table,
+        '3': generate_letters,
+        '4': quit
+    }
+    try:
+        if not arg_dict.get(selection):
+            raise KeyError
```

Comment:
Raising `KeyError` is not necessary here, since it will be raised by Python if `selection` is not a key in the dict in the call to `arg_dict.get(selection)()`

Reply:
to clarify, `arg_dict[selection]` would correctly raise the `KeyError`